Add Forms

steveWDamesJr opened the pull request:
## Great Views

Hello Reviewer 👋 
:sunglasses: This is @steveWDamesJr 's  sixth milestone for the Blog App build. :sunglasses:

In this milestone, I was able to Add the following:
- [x] -Create current_user method
- [x] -Create post on behalf of current user - with validations taken off from 

### Debugging

1. When New post is created,  a debug of the @post variable within the new new.html.erb file in the views and post directory, shows the following:

<img width="1307" alt="image" src="https://user-images.githubusercontent.com/81635839/179372309-c0c4e482-28d1-4f83-a4ba-b7a99fecb4b2.png">

2. However, with validations for post_counter and likes_counter commented out, The post is created : without updating post and like counters.

<img width="879" alt="image" src="https://user-images.githubusercontent.com/81635839/179372716-21a83059-5b2b-4358-b9dd-b33254b9f121.png">

3. Post are successfully created however in rails console:

<img width="911" alt="image" src="https://user-images.githubusercontent.com/81635839/179372767-1fe466f3-c2a2-438a-a34c-dbc5effcbd17.png">

4. Post are created using the following format:
-  user = User.find(4)
- "Post.create(author: user, title: 'Racing Journey', text: 'It is great to be an international driver', comments_counter: 0, likes_counter: 0)"

- [x] -Create comment on behalf of current user
- [x] -Add like functionality for users
- [x] -Debug linters
   
![Our Blog](https://media.giphy.com/media/3coOndBT075fVmxaS7/giphy.gif)



Thank you for taking the time to review my PR :pray:

steveWDamesJr committed:
Add Forms

-Create current_user method
-Create post on behalf of current user - with validations taken off

-Create comment on behalf of current user
-Add like functionality for users
-Debug linter

codepantha commented on db/migrate/20220705164039_create_posts.rb:
```diff
@@ -3,8 +3,8 @@ def change
-      t.integer :comments_counter, default: 0
-      t.integer :likes_counter, default: 0
+      t.integer :comments_counter
+      t.integer :likes_counter
```
![image](https://user-images.githubusercontent.com/81635839/179372309-c0c4e482-28d1-4f83-a4ba-b7a99fecb4b2.png)
- [ ] The reason you are getting the errors you're getting when attempting to create a new post is because you have not set a default value for the `comments_counter` and `likes_counter`.  When a new post is created, logically, such post doesn't have any likes or comments yet. If you fix this, the problems will disappear.

**Hint**
```
t.integer :comments_counter, default: 0
t.integer :likes_counter, default: 0
```

codepantha commented on db/migrate/20220705155001_create_users.rb:
```diff
@@ -4,7 +4,7 @@ def change
       t.string :photo
       t.text :bio
-      t.integer :posts_counter, default: 0
+      t.integer :posts_counter
 
       t.timestamps
```
- [ ] Please reference my earlier suggestion and add a default value of 0 to the `posts_counter`

codepantha reviewed (changes requested):
# STATUS: REQUIRED CHANGES ♻️ 

## Hi @steveWDamesJr 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://media3.giphy.com/media/ZFZLIuWWkr29y/giphy.gif"/>

### Highlights 🎯 
✔️ Descriptive PR
✔️ Correct GitFlow
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

steveWDamesJr committed:
Reinitialize app

steveWDamesJr committed:
Add Forms

-Create current_user method
-Create post on behalf of current user - with validations taken off

-Create comment on behalf of current user
-Add like functionality for users
-Debug linter
-Make comments_counter, likes_counter, posts_counter default to 0
